python/core/auto_generated/metadata/qgsabstractlayermetadataprovider.sip.in

```diff
@@ -0,0 +1,119 @@
+/************************************************************************
+ * This file has been generated automatically from                      *
+ *                                                                      *
+ * src/core/metadata/qgsabstractlayermetadataprovider.h                 *
+ *                                                                      *
+ * Do not edit manually ! Edit header and run scripts/sipify.pl again   *
+ ************************************************************************/
+
+
+
+
+
+
+struct QgsLayerMetadataProviderResult
+{
+  QgsLayerMetadata metadata;
+  QgsPolygon geographicExtent;
+  QgsWkbTypes::GeometryType geometryType;
+  QString crs;
+  QString uri;
+  QString dataProviderName;
+  QgsMapLayerType layerType;
+  QString standardUri;
+
+
+  QString identifier() const;
+%Docstring
+Metadata identifier
+%End
+  QString title() const;
+%Docstring
+Metadata title
+%End
+  QString abstract() const;
+%Docstring
+Metadata abstract
+%End
+};
+
+class QgsLayerMetadataSearchResult
```

Comment:
```suggestion
class QgsLayerMetadataSearchResults
```
?

Reply:
It is a single result record... but I can change it if you think it's best

Reply:
Unless I'm misreading things, doesn't it contain a list of records? (And in future if we have api which collates results from a range of different providers and want to reuse this class then the assumption that it's a single record because its from one provider won't hold)

Reply:
Oh yes, sorry, I confused it with QgsLayerMetadataProviderResult, I'll make it plural.